### PR TITLE
enh(event logs): add possibility to display ack dt comments notifications event handler

### PR DESCRIPTION
**Pull request by cbadr-ext**
## Description

add a filter for ACK et NOTIF rather than teaking the request on other parameters.

CSV export should not been forgotten

**Fixes** # (MON-14623)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [ ] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

add a filter for ACK et NOTIF

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/include/eventLogs/export/Formatter.php [hunk 1]
```diff
@@ -58,6 +58,8 @@ class Formatter
     private string $warning = '';
     private string $critical = '';
     private string $unknown = '';
+    private string $ack = '';
+    private string $notif = '';
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/export/Formatter.php:62]:
```suggestion
    private string $acknowledgement = '';
    private string $notifification = '';
```

Please avoid shortname into variable name to keep it understandable

File: centreon/www/include/eventLogs/export/Formatter.php [hunk 2]
```diff
@@ -167,6 +169,23 @@ public function setUnknown(string $unknown): void
...
+    }
+
+    /**
+     * @param string $notif
+     * @return void
+     */
+    public function setNotif(string $notif): void
+    {
+        $this->notif = $notif;
+    }
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/export/Formatter.php:188]: same here

File: centreon/www/include/eventLogs/export/Formatter.php [hunk 3]
```diff
@@ -192,8 +211,8 @@ public function formatMetaData(): array
             ['Host', 'Up', 'Down', 'Unreachable'],
             ['', $this->up, $this->down, $this->unreachable],
             [],
-            ['Service', 'Ok', 'Warning', 'Critical', 'Unknown'],
-            ['', $this->ok, $this->warning, $this->critical, $this->unknown],
+            ['Service', 'Ok', 'Warning', 'Critical', 'Unknown', 'Ack', 'Notif'],
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/export/Formatter.php:214]:
```suggestion
            ['Service', 'Ok', 'Warning', 'Critical', 'Unknown', 'Acknowledgement', 'Notification'],
```

File: centreon/www/include/eventLogs/export/QueryGenerator.php [hunk 4]
```diff
@@ -46,6 +46,8 @@ class QueryGenerator
     private string $warning;
     private string $critical;
     private string $unknown;
+    private string $ack;
+    private string $notif;
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/export/QueryGenerator.php:50]: same here

File: centreon/www/include/eventLogs/export/QueryGenerator.php [hunk 5]
```diff
@@ -320,6 +322,22 @@ public function setLimit(?int $limit): void
...
+    }
+
+    /**
+     * @param string $notif
+     */
+    public function setNotif(string $notif): void
+    {
+        $this->notif = $notif;
+    }
+
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/export/QueryGenerator.php:340]: same here

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

File: centreon/www/include/eventLogs/xml/data.php [hunk 6]
```diff
@@ -428,6 +440,12 @@
 if ($unknown == 'true') {
     array_push($svc_msg_status_set, "'" . STATUS_UNKNOWN . "'");
 }
+if ($acknowledgement == 'true') {
+    array_push($svc_msg_status_set, "'" . STATUS_ACKNOWLEDGEMRNT . "'");
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/xml/data.php:444]:
```suggestion
    array_push($svc_msg_status_set, "'" . STATUS_ACKNOWLEDGEMENT . "'");
```

File: centreon/www/include/eventLogs/xml/data.php [hunk 7]
```diff
@@ -316,6 +323,8 @@
     STATUS_WARNING => 'service_warning',
     STATUS_CRITICAL => 'service_critical',
     STATUS_UNKNOWN => 'service_unknown',
+    STATUS_ACKNOWLEDGEMRNT => 'service_acknowledgement',
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/xml/data.php:326]:
```suggestion
    STATUS_ACKNOWLEDGEMENT => 'service_acknowledgement',
```

File: centreon/www/include/eventLogs/export/Request.php [hunk 8]
```diff
@@ -442,6 +456,12 @@ public function getSvcMsgStatusSet(): array
         if ($this->getUnknown() === 'true') {
             $this->svcMsgStatusSet[] = sprintf("'%s'", self::STATUS_UNKNOWN);
         }
+        if ($this->getAck() === 'true') {
+            $this->svcMsgStatusSet[] = sprintf("'%s'", self::STATUS_ACKNOWLEDGEMENT);
+        }
+        if ($this->getNotif() === 'true') {
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/export/Request.php:462]:
```suggestion
        if ($this->getNotification() === 'true') {
```

PHP Error: Undefined method getNotif

File: centreon/www/include/eventLogs/export/Formatter.php [hunk 9]
```diff
@@ -167,6 +168,15 @@ public function setUnknown(string $unknown): void
         $this->unknown = $unknown;
     }
 
+    /**
+     * @param string $acknowledgement
+     * @return void
+     */
+    public function setAck(string $acknowledgement): void
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/export/Formatter.php:175]:
```suggestion
    public function setAcknowledgement(string $acknowledgement): void
```

File: centreon/www/include/eventLogs/export/QueryGenerator.php [hunk 10]
```diff
@@ -320,6 +321,14 @@ public function setLimit(?int $limit): void
         $this->limit = $limit;
     }
 
+    /**
+     * @param string $acknowledgement
+     */
+    public function setAck(string $acknowledgement): void
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/export/QueryGenerator.php:327]:
```suggestion
    public function setAcknowledgement(string $acknowledgement): void
```

File: centreon/www/include/eventLogs/export/Request.php [hunk 11]
```diff
@@ -310,6 +312,18 @@ public function getCritical(): string
         return htmlentities($this->critical);
     }
 
+    /**
+     * @return string
+     */
+    public function getAck(): string
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/export/Request.php:318]:
```suggestion
    public function getAcknowledgement(): string
```

File: centreon/www/include/eventLogs/export/Request.php [hunk 12]
```diff
@@ -442,6 +456,12 @@ public function getSvcMsgStatusSet(): array
         if ($this->getUnknown() === 'true') {
             $this->svcMsgStatusSet[] = sprintf("'%s'", self::STATUS_UNKNOWN);
         }
+        if ($this->getAck() === 'true') {
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/export/Request.php:459]:
```suggestion
        if ($this->getAcknowledgement() === 'true') {
```

File: centreon/www/include/eventLogs/export/data.php [hunk 13]
```diff
@@ -110,6 +110,7 @@
 $queryGenerator->setTabSvc($requestHandler->getTabSvc());
 $queryGenerator->setSearchService($requestHandler->getSearchService());
 $queryGenerator->setExport($requestHandler->getExport());
+$queryGenerator->setAck($requestHandler->getAck());
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/export/data.php:113]:
```suggestion
$queryGenerator->setAcknowledgement($requestHandler->getAck());
```

File: centreon/www/include/eventLogs/xml/data.php [hunk 14]
```diff
@@ -84,12 +84,15 @@
 define("STATUS_CRITICAL", 2);
 define("STATUS_UNKNOWN", 3);
 define("STATUS_PENDING", 4);
+define("STATUS_ACKNOWLEDGEMRNT", 5);
```
Comment (jeremyjaouen) [inline centreon/www/include/eventLogs/xml/data.php:87]:
```suggestion
define("STATUS_ACKNOWLEDGEMENT", 5);
```

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by jeremyjaouen — APPROVED** [review]

File: centreon/www/include/eventLogs/export/Request.php [hunk 15]
```diff
@@ -56,7 +56,7 @@ class Request
     private string $warning = 'true';
     private string $critical = 'true';
     private string $unknown = 'true';
-    private string $notification = 'false';
+    private string $acknowledgement = 'true';
```
Comment (callapa) [inline centreon/www/include/eventLogs/export/Request.php:59]: Why delete the $notification variable?

**Review by jeremyjaouen — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Pull request merged**